src/main/java/net/minecraftforge/registries/ForgeRegistry.java

```diff
@@ -234,26 +236,87 @@ public ResourceLocation getKey(V value)
         return getResourceKey(value).map(ResourceKey::location).orElse(this.defaultKey);
     }
 
+    @NotNull
     @Override
     public Optional<ResourceKey<V>> getResourceKey(V value)
     {
         // We use 'owners' here because we want to return the key for the inactive overridden items, not just the active set.
         return Optional.ofNullable(this.owners.inverse().get(value)).map(OverrideOwner::key);
     }
 
+    @SuppressWarnings("unchecked")
+    @Nullable
+    Registry<V> getWrapper()
+    {
+        if (!this.builder.getHasWrapper())
+            return null;
+
+        return this.defaultKey != null
+                ? this.getSlaveMap(NamespacedDefaultedWrapper.Factory.ID, NamespacedDefaultedWrapper.class)
+                : this.getSlaveMap(NamespacedWrapper.Factory.ID, NamespacedWrapper.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    @NotNull
+    NamespacedHolderHelper<V> getHolderHelper()
+    {
+        if (!this.builder.getHasWrapper())
+            throw new IllegalStateException("Cannot query holder helper for non-wrapped registry");
+
+        return this.defaultKey != null
+                ? this.getSlaveMap(NamespacedDefaultedWrapper.Factory.ID, NamespacedDefaultedWrapper.class).getHolderHelper()
+                : this.getSlaveMap(NamespacedWrapper.Factory.ID, NamespacedWrapper.class).getHolderHelper();
```

Comment:
This looks like it could be "simplified" (reduce the long slave map calls) to:
```java
Registry<V> wrapper = getWrapper();
if (wrapper == null)
    throw new IllegalStateException("Cannot query holder helper for non-wrapped registry");
return wrapper.getHolderHelper();
```

And then the suppress warnings for unchecked could probably be removed as well.

Reply:
This is not possible because the two slave maps have no common super type with getHolderHelper() declared